processor-api/src/main/java/com/synopsys/integration/alert/processor/api/NotificationProcessorV2.java

```diff
@@ -92,6 +95,7 @@ private void processAndDistribute(List<AlertNotificationModel> notifications, Co
             ProviderMessageHolder providerMessageHolder = processJobNotifications(jobNotificationWrapper.getProcessingType(), filteredNotifications);
 
             providerMessageDistributor.distribute(processedNotificationDetails, providerMessageHolder);
+            notificationAccessor.setNotificationsProcessedById(notificationIds);
```

Comment:
It might be better to return the notificationIds of the notifications processed (and provide a JavaDoc comment about the return). Otherwise we would need this to be `Transactional` which would lock the DB for the entire duration of notification processing (this seems wrong).